### PR TITLE
rgw/s3select: s3select json output

### PR DESCRIPTION
**Pull request by albin-antony**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by adamemerson** [comment]
jenkins test ceph api

File: src/rgw/rgw_s3select.cc [hunk 1]
```diff
@@ -427,6 +427,9 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_csv(const char* query, const char*
     csv.use_header_info=true;
   }
 
+  if(m_outputFormat == OutputFormat::JSON) {
```
Comment (adamemerson) [inline src/rgw/rgw_s3select.cc:430]: nit: `if` is not a function, put a space before the opening parenthesis, please.

File: src/rgw/rgw_s3select.cc [hunk 2]
```diff
@@ -478,6 +481,10 @@ int RGWSelectObj_ObjStore_S3::run_s3select_on_parquet(const char* query)
   if (!m_s3_parquet_object.is_set()) {
     //parsing the SQL statement.
     s3select_syntax.parse_query(m_sql_query.c_str());
+    parquet_object::csv_definitions parquet;
+    if(m_outputFormat == OutputFormat::JSON) {
+    parquet.output_json_format = true;
```
Comment (adamemerson) [inline src/rgw/rgw_s3select.cc:486]: nit: This should be indented

**Comment by adamemerson** [comment]
jenkins test api

**Comment by adamemerson** [comment]
@albin-antony Please fill out the checklist. Is this a new feature? If so it should have a test.

**Comment by adamemerson** [comment]
Try mentioning the submodule update in the body of the commit message

**Comment by adamemerson** [comment]
Never mind, I see the test in https://github.com/ceph/ceph/pull/58651

**Comment by galsalomon66** [comment]
> Never mind, I see the test in #58651

Thanks
it will be fixed #58651

**Comment by albin-antony** [comment]
> @albin-antony Please fill out the checklist. Is this a new feature? If so it should have a test.

Thanks @adamemerson. Yes, this is a new feature. I have addressed the comments.

**Comment by adamemerson** [comment]
jenkins retest this please

**Comment by galsalomon66** [comment]
this PR is closed, and continues with #58651

**Pull request closed without merging**